src/Tools/Common/ReversedServerHelpers/ReversedServerHelpers.cs

```diff
@@ -0,0 +1,112 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.NETCore.Client;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Internal.Common.Utils
+{
+    // <summary>
+    // ProcessLauncher is a child-process launcher for "diagnostics tools at startup" scenarios
+    // It launches the target process at startup and passes its processId to the corresponding Command handler.
+    // </summary>
+    internal class ProcessLauncher
+    {
+        private Process _childProc;
+        public ManualResetEvent HasExited;
+
+        internal static ProcessLauncher Launcher = new ProcessLauncher();
+
+        public void PrepareChildProcess(List<string> args)
+        {
+            _childProc = new Process();
+            _childProc.StartInfo.FileName = args[0];
+            _childProc.StartInfo.Arguments = String.Join(" ", args.GetRange(1, args.Count - 1));
```

Comment:
I don't think we can use that syntax because the tools are on netcoreapp2.1 which doesn't have C# 8 features...?

Reply:
Sadness, I love the new slice syntax

Reply:
While it might be a bit too early...

2.1 goes out of support in August.  We may want to move dotnet diagnostic tools to 3.1 before then.

I'd be half tempted to move to 3.1 as soon as we finish the 5.0 release.

Reply:
I think we can start the discussion on that after we finish 5.0 release. I filed https://github.com/dotnet/diagnostics/issues/1640 to keep track of that work.